File: 2021/day-2/part-2/main.py
def main() -> int:
    horiz, depth, aim = 0, 0, 0
    data = open("inputs.txt").readlines()
    for item in data:
        text, number = item.split(" ")
        match text:
            case "forward":
                horiz += int(number)
                depth += aim*int(number)
            case "down":
                aim += int(number)
            case "up":
                aim -= int(number)
    return horiz * depth

if __name__ == "__main__":
    print(main())

File: 2021/day-2/part-2/main.jl
function main()
    horiz, depth, aim = 0, 0, 0
    data = readlines(open("inputs.txt", "r"))
    for item in data
        text, number = split(item, " ")
        if text == "forward"
            horiz += parse(Int, number)
            depth += aim*parse(Int, number)
        elseif text == "down"
            aim += parse(Int, number)
        elseif text == "up"
            aim -= parse(Int, number)
        end
    end
    return horiz * depth
end
println(main())